polynote-kernel/src/main/scala/polynote/kernel/util/ManagedMap.scala

```diff
@@ -0,0 +1,69 @@
+package polynote.kernel.util
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+import zio.{Cause, Exit, Promise, Reservation, Semaphore, UIO, ZIO, ZManaged}
+import ZIO.{effect, effectTotal}
+
+class ManagedMap[K, V] private (
```

Comment:
Not using this yet, will be using to transition things like `NotebookManager` and `Repository` to `ZManaged` (in place of `RefMap`)